### PR TITLE
make llm completion program more general

### PR DESCRIPTION
**Pull request by jerryjliu**
before, it only took a pydantic parser

however, other output parsers could also return pydantic objects. let users specify any output parser, and add the option to additionally specify an output class

### CODE REVIEW

File: llama_index/program/llm_program.py [hunk 1]
```diff
@@ -47,16 +50,24 @@ def from_defaults(
             raise ValueError("Must provide either prompt or prompt_template_str.")
         if prompt_template_str is not None:
             prompt = PromptTemplate(prompt_template_str)
+
+        # decide default output class if not set
+        if output_cls is None:
```
Comment (logan-markewich) [inline llama_index/program/llm_program.py:55]: much needed!

Comment (logan-markewich) [reply llama_index/program/llm_program.py:55]: Should it just default to a pydantic output parser if not provided?

Comment (jerryjliu) [reply llama_index/program/llm_program.py:55]: you mean if output_cls is not None but output parser is?

Comment (logan-markewich) [reply llama_index/program/llm_program.py:55]: Yea. That would be more similar to the openai pydantic program. Less imprts for the user

**Pull request merged**